In [ ]:
import pandas as pd
import numpy as np
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

print(df.isnull().sum())

print(len(df))

dfPost = df.dropna(subset=['Android Ver', 'Type'])

print(len(dfPost))

df['Rating'] = df['Rating'].fillna(df.groupby('Category')['Rating'].transform('median'))

df['Size'] = df['Size'].replace("Varies with device", np.nan)

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64
10841
10837


Pregunta A: Ejecuten df.isnull().sum(). Identificarán que la columna Rating tiene casi 1,500 valores nulos, mientras que Type y Android Ver tienen solo 1 o 2. ¿Por qué es arquitectónicamente seguro eliminar las filas donde falta Android Ver, pero sería un error fatal eliminar las filas donde falta Rating?

Podriamos eliminar la version de android ya que estos realmente no aportarian tanto a lo que es un analisis en cuanto a los datos, donde si seria un error es en rating, ya que estos casos realmente existen y no es por motivos aleatorios o errores, los usuarios tienden a no dejar reseñas en aplicaciones, y esto puede ayudar a un modelo a aprender sobre ello

Pregunta B: Escriban la instrucción exacta usando dropna() pasándole el parámetro subset=['Android Ver', 'Type'] para eliminar únicamente las 3 filas que tienen esos nulos. Verifiquen que el total de registros en el DataFrame apenas disminuyó.

Bajaron 4 registros, de 10841 a 10837, ya que fueron 3 de Android Ver y 1 de Type

Pregunta C: Rellenar los 1,500 nulos de Rating con el promedio global es un error, ya que las apps de la categoría FAMILY (Familia) no se califican igual que las de DATING (Citas). Utilizando la lógica de la semana pasada, escriban un bloque combinando .groupby('Category') y .transform() para rellenar los NaN de la columna Rating usando la mediana de su propia categoría.

Yo veo correcto rellenar las areas por seccion, asi se deja de manera mas pareja por cada una

Pregunta D: Revisen la columna Size (Tamaño de la app). Notarán que no tiene nulos matemáticos (NaN), pero muchas filas dicen "Varies with device". Para un modelo matemático, ese texto es basura indigerible. Escriban una instrucción usando df['Size'].replace(...) para convertir exactamente esa frase de texto en un nulo matemático real usando np.nan de NumPy. ¿Por qué es mejor convertirlo a NaN en esta etapa en lugar de ponerle un tamaño falso como "0 MB"?

Ponerle un tamaño falso como ese de 0 MB habilitaria un sesgo inferior, se;alando que hay valores como 0 cuando se sabe que es imposible, pero dejandolos en un NaN se pueden ignorar estos datos que da;arian mas de lo que ayudarian al modelo